test/DynamoCoreTests/PackageDependencyTests.cs

```diff
@@ -457,17 +457,53 @@ public void PackageDependencyStateTest()
             // but the nodes are resolved by a different package.
             PackageDependencyInfo secondPackage = (PackageDependencyInfo) packageDependenciesList[1];
             Assert.AreEqual(new PackageDependencyInfo("Clockwork for Dynamo 2.x", new Version("2.1.2")), secondPackage);
-            Assert.AreEqual(PackageDependencyState.Warning, secondPackage.State);
+            Assert.AreEqual(PackageDependencyState.Missing, secondPackage.State);
 
             // Check for Missing package state
             PackageDependencyInfo thirdPackage = (PackageDependencyInfo) packageDependenciesList[2];
             Assert.AreEqual(new PackageDependencyInfo("Clockwork for Dynamo 1.x", new Version("1.33.0")), thirdPackage);
-            Assert.AreEqual(PackageDependencyState.IncorrectVersion, thirdPackage.State);
+            Assert.AreEqual(PackageDependencyState.Missing, thirdPackage.State);
 
             // Check for Loaded package state
             PackageDependencyInfo lastPackage = (PackageDependencyInfo) packageDependenciesList.Last();
             Assert.AreEqual(new PackageDependencyInfo("Dynamo Samples", new Version("2.0.0")), lastPackage);
             Assert.AreEqual(PackageDependencyState.Loaded, lastPackage.State);
         }
+
+        [Test]
+        public void PackageDependencyStatechangeTestAfterLoadingPackage()
+        {
+            string packageDirectory = Path.Combine(TestDirectory, @"core\packageDependencyTests\ClockworkPackage");
+            LoadPackage(packageDirectory);
+
+            // Load JSON file graph
+            string path = Path.Combine(TestDirectory, @"core\packageDependencyTests\PackageDependencyStates.dyn");
+            OpenModel(path);
+
+            // Assert the total number of package dependencies.
+            var packageDependenciesList = CurrentDynamoModel.CurrentWorkspace.NodeLibraryDependencies;
+            Assert.AreEqual(4, packageDependenciesList.Count);
+
+            // Check for Incorrect package state
+            PackageDependencyInfo firstPackage = (PackageDependencyInfo)packageDependenciesList[0];
+            Assert.AreEqual(new PackageDependencyInfo("MeshToolkit", new Version("2.0.1")), firstPackage);
+            Assert.AreEqual(PackageDependencyState.Missing, firstPackage.State);
+
+            // Check for Warning package state, where the actually package is missing
+            // but the nodes are resolved by a different package.
+            PackageDependencyInfo secondPackage = (PackageDependencyInfo)packageDependenciesList[1];
+            Assert.AreEqual(new PackageDependencyInfo("Clockwork for Dynamo 2.x", new Version("2.1.2")), secondPackage);
+            Assert.AreEqual(PackageDependencyState.Warning, secondPackage.State);
+
+            // Check for Missing package state
+            PackageDependencyInfo thirdPackage = (PackageDependencyInfo)packageDependenciesList[2];
```

Comment:
@reddyashish Did you figure if you could call `LoadPackage` in the middle of the test for the missing case?

Reply:
The state doesn't change then because the file has to be reopened again to get the updated data.

Reply:
Updated the test to check for the state change in the same test. Dynamo restart is not needed.